### PR TITLE
Cancel auctions on Coretime launch

### PR DESCRIPTION
**Pull request by eskimor**
Parachain Auctions that would conclude after Coretime launch, would not be served, yet we are cancelling all auctions that are still ongoing. 

- [ ] Does not require a CHANGELOG entry

### CODE REVIEW

**Comment by bkchr** [comment]
> * Does not require a CHANGELOG entry

IMO it should be added to the Changelog.

File: relay/kusama/src/lib.rs [hunk 1]
```diff
@@ -1849,6 +1851,23 @@ pub mod migrations {
...
+	///
+	/// Any leases that come into existance, after coretime was launched will not be served. Yet,
+	/// any still ongoing auctions must be cancelled.
+	///
+	/// Safety: After coretime is launched, there are no auctions anymore. So if this forgotten to
+	/// be removed after the runtime upgrade, running this again on the next one is harmless.
+	pub struct CancelAuctions;
+	impl OnRuntimeUpgrade for CancelAuctions {
+		fn on_runtime_upgrade() -> Weight {
+			if let Err(err) = Auctions::cancel_auction(frame_system::RawOrigin::Root.into()) {
```
Comment (bkchr) [inline relay/kusama/src/lib.rs:1864]: What about the scheduled auctions? I see there is currently one auction scheduled. Generally not sure if we always scheduled them manually?

Comment (eskimor) [reply relay/kusama/src/lib.rs:1864]: Good point!

Comment (eskimor) [reply relay/kusama/src/lib.rs:1864]: Ok, where are they handled?

Comment (joepetrowski) [reply relay/kusama/src/lib.rs:1864]: They should be under the Scheduler pallet's `Agenda`. We will just have to kill any items that are auctions.

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Feature for enabling debug prints in the Polkadot and Kusama runtime ([polkadot-fellows/runtimes#85](https://github.com/polkadot-fellows/runtimes/pull/85))
 - Added new "Wish for Change" track ([polkadot-fellows/runtimes#184](https://github.com/polkadot-fellows/runtimes/pull/184))
 - Enable Coretime and on-demand on Kusama ([polkadot-fellows/runtimes#159](https://github.com/polkadot-fellows/runtimes/pull/159)
+- Cancel Parachain Auctions [polkadot-fellows/runtimes#215](https://github.com/paritytech/polkadot-sdk/pull/2226)
```
Comment (joepetrowski) [inline CHANGELOG.md:17]:
```suggestion
- Cancel Parachain Auctions [polkadot-fellows/runtimes#215](https://github.com/polkadot-fellows/runtimes/pull/215)
```

File: relay/kusama/src/lib.rs [hunk 3]
```diff
@@ -1849,6 +1852,41 @@ pub mod migrations {
 		}
 	}
 
+	/// Cancel all ongoing auctions.
+	///
+	/// Any leases that come into existance, after coretime was launched will not be served. Yet,
+	/// any still ongoing auctions must be cancelled.
```
Comment (joepetrowski) [inline relay/kusama/src/lib.rs:1858]:
```suggestion
	/// Any leases that come into existence after coretime was launched will not be served. Yet,
	/// any ongoing auctions must be cancelled.
```

File: relay/kusama/src/lib.rs [hunk 4]
```diff
@@ -1849,6 +1852,41 @@ pub mod migrations {
...
+	///
+	/// Safety:
+	///
+	/// - After coretime is launched, there are no auctions anymore. So if this forgotten to
+	/// be removed after the runtime upgrade, running this again on the next one is harmless.
+	/// - I am assuming scheduler `TaskName`s are unique, so removal of the scheduled entry
+	/// multiple times should also be fine.
```
Comment (joepetrowski) [inline relay/kusama/src/lib.rs:1865]: Yes, they are unique.
https://github.com/paritytech/polkadot-sdk/blob/polkadot-v1.7.2/substrate/frame/scheduler/src/lib.rs#L895

**Review by joepetrowski — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Review by ordian — APPROVED** [review]

File: relay/kusama/src/lib.rs [hunk 5]
```diff
@@ -1849,6 +1852,41 @@ pub mod migrations {
...
+	pub struct CancelAuctions;
+	impl OnRuntimeUpgrade for CancelAuctions {
+		fn on_runtime_upgrade() -> Weight {
+			if let Err(err) = Auctions::cancel_auction(frame_system::RawOrigin::Root.into()) {
+				log::error!("Cancelling auctions failed: {:?}", err);
+			}
+			// Cancel scheduled auction as well:
+			if let Err(err) = Scheduler::cancel_named(
+				pallet_custom_origins::Origin::AuctionAdmin.into(),
```
Comment (eskimor) [inline relay/kusama/src/lib.rs:1874]: This does not seem to work. I am getting:

```
fellowship-runtimes/relay/kusama/src/lib.rs:1871:96:
    called `Result::unwrap()` on an `Err` value: Module(ModuleError { index: 29, error: [1, 0, 0, 0], message: Some("NotFound") })
```
Does not work with `cancel` directly either. :thinking:

Comment (eskimor) [reply relay/kusama/src/lib.rs:1874]: It gets executed multiple times with try-runtime - that's where this error is coming from.

File: relay/kusama/src/lib.rs [hunk 6]
```diff
@@ -1849,6 +1852,41 @@ pub mod migrations {
...
+	impl OnRuntimeUpgrade for CancelAuctions {
+		fn on_runtime_upgrade() -> Weight {
+			if let Err(err) = Auctions::cancel_auction(frame_system::RawOrigin::Root.into()) {
+				log::debug!(target: "runtime", "Cancelling auctions failed: {:?}", err);
+			}
+			// Cancel scheduled auction as well:
+			if let Err(err) = Scheduler::cancel_named(
+				pallet_custom_origins::Origin::AuctionAdmin.into(),
+				[
+					0x5c, 0x68, 0xbf, 0x0c, 0x2d, 0x11, 0x04, 0x91, 0x6b, 0xa5, 0xa4, 0xde, 0xe6,
```
Comment (bkchr) [inline relay/kusama/src/lib.rs:1876]: Trusting you on this name here ;)

**Review by bkchr — APPROVED** [review]

**Pull request merged**